pkg/wallet/api.go

```diff
@@ -55,49 +56,29 @@ type Crypto interface {
 	// error: error
 	SignMessage(message []byte, fromVerKey string) ([]byte, error)
 
-	// DecryptMessage decrypt message
-	//
-	// Args:
-	//
-	// encMessage: The encrypted message content
-	//
-	// toVerKey:The verification key of the recipient.
-	//
-	// []byte: Decrypted message content
-	//
-	// string: The sender verification key
-	//
-	// error: error
-	DecryptMessage(encMessage []byte, toVerKey string) ([]byte, string, error)
-}
-
-// Pack provide methods to pack and unpack msg
-type Pack interface {
-	// PackMessage Pack a message for one or more recipients.
-	//
-	// Args:
-	//
-	// envelope: The message to pack
-	//
-	// Returns:
-	//
-	// []byte: The packed message
-	//
-	// error: error
-	PackMessage(envelope *Envelope) ([]byte, error)
-
-	// UnpackMessage Unpack a message.
-	//
-	// Args:
-	//
-	// encMessage: The encrypted message
-	//
-	// Returns:
-	//
-	// envelope: unpack message
-	//
-	// error: error
-	UnpackMessage(encMessage []byte) (*Envelope, error)
+	// DeriveKEK will derive an ephemeral symmetric key (kek) using a private key and toPubKey to
+	// be used for encrypting a cek.
+	//
+	// This function assumes both fromKey and toPubKey to be on curve25519.
+	//
+	// isFromKeyPriv flag is to direct the function to use fromKey as is if set to true
+	// 		(ie fromKey is an ephemeral private key provided by the caller)
+	// or will fetch the private key corresponding to fromKey from the wallet if set to false
+	// 		(ie fromKey is a public key from the wallet and requires retrieving its private key. This
+	// 		function is part of the wallet for this reason)
+	// returns:
+	// 		kek []byte
+	//		error in case of errors
+	DeriveKEK(alg, apu []byte, fromKey, toPubKey *[chacha.KeySize]byte, isFromKeyPriv bool) ([]byte, error)
```

Comment:
Is there a case where private key would be provided here? I am not clear why we need to expose isFromKeyPriv flag here.

Reply:
yes, when deriving a KEK from an ephemeral key created in the crypter, in this case `fromKey` will be a private key.. this is why we need `isFromKeyPriv` flag

but deriving a KEK for a CEK doesn't require an ephemeral key, and therefore `fromKey` will be public to tell the function to load its corresponding private key from the wallet

Reply:
I can probably hide this function and expose 2 functions in the interface:
DeriveKEKUsingPrivKeyFromWallet() which will call `DeriveKEK` with `isFromKeyPriv`= false
DeriveKEKUsingEphemeralPrivKey() which will call `DeriveKEK` with `isFromKeyPriv`= true

this way `isFromKeyPriv` flag is hidden from the interface user. 

Agree?

Reply:
Does it make more sense to create utility function DeriveKEK that will work with private key? In that case we can have have only one method in the wallet that will retrieve private key and call utility function DeriveKEK. And for ephemeral keys since you already have private key you just call utility function DeriveKEK. I am not sure where to put this utility function though.

Reply:
I'll move the util function to util.go under wallet package.. will push soon

Reply:
member DeriveKEK function takes fromKey as public key and calls a util exported function with the same name that does the real derivation using a private key.

done